anndata/_core/sparse_dataset.py

```diff
@@ -407,11 +413,11 @@ def __getitem__(self, index: Index | tuple[()]) -> float | ss.spmatrix:
         mtx = self._to_backed()
 
         # Handle masked indexing along major axis
-        if self.format == "csr" and np.array(row).dtype == bool:
+        if self.format == "csr" and np.array(row).dtype == bool and row.sum() != 0:
```

Comment:
Why do `sum` here? Can't we tell this from the result of `ezclump`? E.g.

```python
In [3]: np.ma.extras._ezclump(np.zeros(10, dtype=bool))
Out[3]: []
```

Reply:
Vibes more than anything - felt weird to add that code here.  Also, I think `sum` is faster by orders of magnitude, no?

```python
import numpy as np
arr = np.random.randint(size=1_000_000, low=0, high=2).astype(bool)
%timeit np.ma.extras._ezclump(arr)
# 29.5 ms ± 353 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
%timeit arr.sum()
# 205 µs ± 1.56 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

Reply:
> Vibes more than anything - felt weird to add that code here since it is unrelated to the rest. Also, I think `sum` is faster by orders of magnitude, no?
> 
> ```python
> import numpy as np
> arr = np.random.randint(size=1_000_000, low=0, high=2).astype(bool)
> %timeit np.ma.extras._ezclump(arr)
> # 29.5 ms ± 353 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
> %timeit arr.sum()
> # 205 µs ± 1.56 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
> ```

Reply:
> Also, I think sum is faster by orders of magnitude, no?

Not when the array is all zero:

```python
arr = np.zeros(1_000_000, dtype=bool)

%timeit arr.sum()
# 367 µs ± 16.2 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

%timeit np.ma.extras._ezclump(arr)
# 746 µs ± 21.7 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```

Still a little slower, but it's a case where performance probably isn't an issue. I'd be fine either way.

More importantly:

It makes more sense to me if this logic lives in `subset_by_major_axis_mask`. I think it would be better if `subset_by_major_axis_mask` didn't error if we pass in an all null array. We know the bug is there now, let's not wait until it goes off.

I would be okay if `subset_by_major_axis_mask` called `sum`, I just want it not to choke on an all False mask.